File: 30-03-2026/truncate_text_2.js
function truncateText(str) {
  let numStr = str
    .replace(/[ilI\.]/g, 1)
    .replace(/[fjrt\s]/g, 2)
    .replace(/[a-zJL]/g, 3)
    .replace(/[A-Z]/g, 4)
    ?.split("")
    ?.map(Number);

  let count = 0;
  for (let i = 0; i < numStr.length || 0; i++) {
    count += numStr[i];

    const nextCount = count + (numStr[i + 1] || 0);

    if (nextCount <= 50 && numStr[i + 1] == undefined) return str;
    if (nextCount + 3 > 50 && numStr[i + 1] != undefined)
      return str.slice(0, i + 1) + "...";
  }
}

console.log(truncateText("The quick brown fox")); // "The quick brown f..."
console.log(truncateText("The silky smooth sloth")); // "The silky smooth s..."
console.log(truncateText("THE LOUD BRIGHT BIRD")); // "THE LOUD BRIG..."
console.log(truncateText("The fast striped zebra")); // "The fast striped z..."
console.log(truncateText("The big black bear")); // "The big black bear"

File: 30-03-2026/truncate_text_2.py
import re

def truncate_text(s):
    firstSub = re.sub(r'[ilI\.]', '1', s)
    secondSub = re.sub(r'[fjrt\s]', '2', firstSub)
    thirdSub = re.sub(r'[a-zJL]', '3', secondSub)
    fourthSub = re.sub(r'[A-Z]', '4', thirdSub)
    numStr = list(map(int, list(fourthSub)))

    count = 0
    for i in range(len(numStr)):
        count += numStr[i]
        nextCount = count + (numStr[i + 1] or 0) if i + 1 < len(numStr) else count
        
        if nextCount <= 50 and i + 1 >= len(numStr):
            return s
        if nextCount + 3 > 50 and i + 1 <= len(numStr):
            return s[: i + 1] + "...";


print(truncate_text("The quick brown fox")); # "The quick brown f..."
print(truncate_text("The silky smooth sloth")); # "The silky smooth s..."
print(truncate_text("THE LOUD BRIGHT BIRD")); # "THE LOUD BRIG..."
print(truncate_text("The fast striped zebra")); # "The fast striped z..."
print(truncate_text("The big black bear")); # "The big black bear"
